### PR TITLE
Rewrite the attack and defend calc logic

### PR DESCRIPTION
**Pull request by ibratoev**
Follow the logic from a 3 years old Polytopia version that was not obfuscated. Since then, Polytopia is obfuscated and the code is hard to be extracted.

### CODE REVIEW

File: bot/commands/units.js [hunk 1]
```diff
@@ -139,6 +139,13 @@ module.exports = {
         boost(unit)
     }
 
+    // Add integer values to be used for damage calculation.
+    unit.iAtt = BigInt(unit.att * 100);
+    unit.iBonus = BigInt(unit.bonus * 10);
+    unit.iDef = BigInt(unit.def * 100 * unit.bonus);
```
Comment (veerjd) [inline bot/commands/units.js:145]: Shouldn't you multiple by unit.iBonus instead of unit.bonus at this point?

Comment (veerjd) [reply bot/commands/units.js:145]: Clearly not, because it works

**Review by veerjd — APPROVED** [review]
I agree that this resolves everything

**Review by veerjd — APPROVED** [review]

File: bot/commands/units.js (hunk 1 above)
Comment (ibratoev) [reply bot/commands/units.js:145]: Yeah, the problem is that you can't mix Numbers and BigInts. I can write it as:
`BigInt(unit.def * 10 * Number(unit.iBonus));` but it is not much better.

Comment (veerjd) [reply bot/commands/units.js:145]: Yeah, no changes needed

**Pull request merged**